clickhouse: 18.14.9 -> 18.14.18

r-ryantm opened the pull request:
Semi-automatic update generated by https://github.com/ryantm/nixpkgs-update tools. This update was made based on information from https://repology.org/metapackage/clickhouse/versions.



meta.description for clickhouse is: '"Column-oriented database management system"'.


[Compare changes on GitHub](https://github.com/yandex/ClickHouse/compare/v18.14.9-stable...v18.14.18-stable)

<details>
<summary>
Checks done (click to expand)
</summary>

- built on NixOS
- Warning: no invocation of /nix/store/vm87ngzyw68dnkvc1iydy5zc8dzpxabr-clickhouse-18.14.18/bin/clickhouse-report had a zero exit code or showed the expected version
- Warning: no invocation of /nix/store/vm87ngzyw68dnkvc1iydy5zc8dzpxabr-clickhouse-18.14.18/bin/clickhouse-server had a zero exit code or showed the expected version
- /nix/store/vm87ngzyw68dnkvc1iydy5zc8dzpxabr-clickhouse-18.14.18/bin/clickhouse-client passed the binary check.
- /nix/store/vm87ngzyw68dnkvc1iydy5zc8dzpxabr-clickhouse-18.14.18/bin/clickhouse-local passed the binary check.
- /nix/store/vm87ngzyw68dnkvc1iydy5zc8dzpxabr-clickhouse-18.14.18/bin/clickhouse-benchmark passed the binary check.
- /nix/store/vm87ngzyw68dnkvc1iydy5zc8dzpxabr-clickhouse-18.14.18/bin/clickhouse-performance-test passed the binary check.
- /nix/store/vm87ngzyw68dnkvc1iydy5zc8dzpxabr-clickhouse-18.14.18/bin/clickhouse-copier passed the binary check.
- /nix/store/vm87ngzyw68dnkvc1iydy5zc8dzpxabr-clickhouse-18.14.18/bin/clickhouse-extract-from-config passed the binary check.
- /nix/store/vm87ngzyw68dnkvc1iydy5zc8dzpxabr-clickhouse-18.14.18/bin/clickhouse-compressor passed the binary check.
- Warning: no invocation of /nix/store/vm87ngzyw68dnkvc1iydy5zc8dzpxabr-clickhouse-18.14.18/bin/clickhouse-format had a zero exit code or showed the expected version
- /nix/store/vm87ngzyw68dnkvc1iydy5zc8dzpxabr-clickhouse-18.14.18/bin/clickhouse-obfuscator passed the binary check.
- /nix/store/vm87ngzyw68dnkvc1iydy5zc8dzpxabr-clickhouse-18.14.18/bin/clickhouse-odbc-bridge passed the binary check.
- /nix/store/vm87ngzyw68dnkvc1iydy5zc8dzpxabr-clickhouse-18.14.18/bin/clickhouse-clang passed the binary check.
- /nix/store/vm87ngzyw68dnkvc1iydy5zc8dzpxabr-clickhouse-18.14.18/bin/clickhouse-lld passed the binary check.
- Warning: no invocation of /nix/store/vm87ngzyw68dnkvc1iydy5zc8dzpxabr-clickhouse-18.14.18/bin/clickhouse-test had a zero exit code or showed the expected version
- Warning: no invocation of /nix/store/vm87ngzyw68dnkvc1iydy5zc8dzpxabr-clickhouse-18.14.18/bin/clickhouse-test-server had a zero exit code or showed the expected version
- Warning: no invocation of /nix/store/vm87ngzyw68dnkvc1iydy5zc8dzpxabr-clickhouse-18.14.18/bin/config-processor had a zero exit code or showed the expected version
- /nix/store/vm87ngzyw68dnkvc1iydy5zc8dzpxabr-clickhouse-18.14.18/bin/corrector_utf8 passed the binary check.
- Warning: no invocation of /nix/store/vm87ngzyw68dnkvc1iydy5zc8dzpxabr-clickhouse-18.14.18/bin/clickhouse-zookeeper-cli had a zero exit code or showed the expected version
- Warning: no invocation of /nix/store/vm87ngzyw68dnkvc1iydy5zc8dzpxabr-clickhouse-18.14.18/bin/clickhouse had a zero exit code or showed the expected version
- 12 of 20 passed binary check by having a zero exit code.
- 2 of 20 passed binary check by having the new version present in output.
- found 18.14.18 with grep in /nix/store/vm87ngzyw68dnkvc1iydy5zc8dzpxabr-clickhouse-18.14.18
- directory tree listing: https://gist.github.com/87160b8da3952522296ee5a3b4b1c19b
- du listing: https://gist.github.com/689dd7f3bea1fe97f0b8d0e7bdd2a248

</details>
<details>
<summary>
Rebuild report (if merged into master) (click to expand)
</summary>

3 total rebuild path(s)

1 package rebuild(s)

1 x86_64-linux rebuild(s)
1 i686-linux rebuild(s)
0 x86_64-darwin rebuild(s)
1 aarch64-linux rebuild(s)


First fifty rebuilds by attrpath
clickhouse

</details>

<details>
<summary>
Instructions to test this update (click to expand)
</summary>

Either download from Cachix:
```
nix-store -r /nix/store/vm87ngzyw68dnkvc1iydy5zc8dzpxabr-clickhouse-18.14.18 \
  --option binary-caches 'https://cache.nixos.org/ https://r-ryantm.cachix.org/' \
  --option trusted-public-keys '
  r-ryantm.cachix.org-1:gkUbLkouDAyvBdpBX0JOdIiD2/DP1ldF3Z3Y6Gqcc4c=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(r-ryantm's Cachix cache is only trusted for this store-path realization.)

Or, build yourself:
```
nix-build -A clickhouse https://github.com/r-ryantm/nixpkgs/archive/4852a6c9522927dff0ddceea9c2eb4d894cb6952.tar.gz
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/vm87ngzyw68dnkvc1iydy5zc8dzpxabr-clickhouse-18.14.18
ls -la /nix/store/vm87ngzyw68dnkvc1iydy5zc8dzpxabr-clickhouse-18.14.18/bin
```


</details>
<br/>


cc @orivej for testing.

r-ryantm committed:
clickhouse: 18.14.9 -> 18.14.18

Semi-automatic update generated by
https://github.com/ryantm/nixpkgs-update tools. This update was made
based on information from
https://repology.org/metapackage/clickhouse/versions

Ma27 commented:
The package itself seems fine, but the module breaks (obviously for a while now) as the package tries to log everything to `/var/log`. Currently working on a fix, I'll push it tomorrow :)

Ma27 committed:
clickhouse: fix module and package runtime

Although the package itself builds fine, the module fails because it
tries to log into a non-existant file in `/var/log` which breaks the
service. Patching to default config to log to stdout by default fixes
the issue. Additionally this is the better solution as NixOS heavily
relies on systemd (and thus journald) for logging.

Also, the runtime relies on `/etc/localtime` to start, as it's not
required by the module system we set UTC as sensitive default when using
the module.

To ensure that the service's basic functionality is available, a simple
NixOS test has been added.

Ma27 commented:
@orivej I added some minor fixes to fix the module. Unless you have anything to add I'd merge tomorrow :)

Mic92 commented on pkgs/servers/clickhouse/default.nix:
```diff
@@ -36,6 +36,11 @@ stdenv.mkDerivation rec {
     rm -rf $out/share/clickhouse-test
+
+    sed -i -e '/<log>\/var\/log\/clickhouse-server\/clickhouse-server\.log<\/log>/d' \
+      $out/etc/clickhouse-server/config.xml
```
you can use `!` to delimite regex-parts to avoid escaping slashes in your paths.

Ma27 replied:
makes sense. I'm currently rebuilding this :)

Ma27 commented:
@Mic92 anything to add from your side? Otherwise I'd merge this PR to master :)

Ma27 commented:
merging for now (as the module is also broken on master).
